src/ci/docker/host-x86_64/x86_64-gnu-llvm-15/Dockerfile

```diff
@@ -11,7 +11,6 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
   file \
   curl \
   ca-certificates \
-  python2.7 \
```

Comment:
both Dockerfiles have an outdated comment about how they use both Python 2 and 3

Reply:
@jyn514 `mingw-check-tidy` still uses it -- should be good enough?
https://github.com/rust-lang/rust/blob/fef27e038ee9886722548b70d0c7f75d7e666c72/src/ci/docker/host-x86_64/mingw-check-tidy/Dockerfile#L34

Reply:
Do we block on that in a full bors merge? I was under the impression it only ran on PRs.

Reply:
Hmm, I suppose that is just a `pr` job. So, I can add something back to llvm-15 if you want, but the new llvm-16 builder is on `ubuntu:23.04` which doesn't have python2 at all. Or we can find some other builder as the guinea pig, or maybe it's time to soft-retire python2 and stop worrying about it...

Reply:
Let's find some other builder to be the guinea pig. I see src/ci/docker/host-x86_64/disabled/dist-i686-android/Dockerfile is using 16.04, would that work?

Reply:
It seems like we're barely testing python2 compat at all, given that `x.py` re-execs when it sees python3 available, but oh well. I've added it to `dist-x86_64-illumos` as suggested.